geoengine/client.py

```diff
@@ -0,0 +1,474 @@
+                      name_filter: Optional[str] = None,
+                      timeout: int = 60) -> List[api.DatasetListing]:
+        '''List datasets'''
+        # pylint: disable=too-many-arguments
+        return datasets.list_datasets(self.get_session(), offset, limit, order, name_filter, timeout)
+
+    def layer_collection(self, layer_collection_id: Optional[layers.LayerCollectionId] = None,
+                         layer_provider_id: layers.LayerProviderId = layers.LAYER_DB_PROVIDER_ID,
+                         timeout: int = 60) -> layers.LayerCollection:
+        '''
+        Retrieve a layer collection that contains layers and layer collections.
+        '''
+        return layers.layer_collection(self.get_session(), layer_collection_id, layer_provider_id, timeout)
+
+    def layer_collection_add_sub_collection(
+            self,
+            layer_collection: layers.LayerCollection,
+            name: str,
+            description: str,
+            timeout: int = 60
+    ) -> layers.LayerCollectionId:
+        ''' Add a sub collection to a layer collection '''
+        return layer_collection.add_collection(self.get_session(), name, description, timeout)
+
+    def layer_collection_add_existing_collection(
+            self,
+            layer_collection: layers.LayerCollection,
+            existing_collection: Union[layers.LayerCollectionListing, layers.LayerCollection, layers.LayerCollectionId],
+            timeout: int = 60
+    ) -> layers.LayerCollectionId:
+        ''' Add an existing collection to a layer collection '''
+        return layer_collection.add_existing_collection(self.get_session(), existing_collection, timeout)
+
+    def layer_collection_add_layer(
+            self,
+            layer_collection: layers.LayerCollection,
+            name: str,
+            description: str,
+            workflow: workflows.WorkflowType,
+            symbology: Optional[layers.Symbology],
+            timeout: int = 60) -> layers.LayerId:
+        '''Add a layer to this collection'''
+        # pylint: disable=too-many-arguments
+        return layer_collection.add_layer(self.get_session(), name, description, workflow, symbology, timeout)
+
+    def layer_collection_add_existing_layer(
+            self,
+            layer_collection: layers.LayerCollection,
+            existing_layer: Union[layers.Layer, layers.LayerId],
+            timeout: int = 60
+    ) -> layers.LayerId:
+        ''' Add an existing layer to a layer collection '''
+        return layer_collection.add_existing_layer(self.get_session(), existing_layer, timeout)
+
+    def layer_listing_resolve(
+            self,
+            listing: layers.Listing,
+            timeout: int = 60
+    ) -> Union[layers.LayerCollection, layers.Layer]:
+        '''
+        Retrieve a layer collection that contains layers and layer collections.
+        '''
+        return listing.load(self.get_session(), timeout)
+
+    def layer_collection_reload(
```

Comment:
does this method still make sense? What is the difference to resolve?

Reply:
the parameters and result types are different. We could change that but we should do that when we decide to implement the client like this